tdrs-backend/tdpservice/parsers/validators.py

```diff
@@ -351,17 +351,31 @@ def isInLimits(LowerBound, UpperBound):
         lambda value: f"{value} is not larger or equal to {LowerBound} and smaller or equal to {UpperBound}.",
     )
 
+def intHasLength(num_digits):
+    """Validate the number of digits in an integer."""
+    return make_validator(
+        lambda value: len(str(value)) == num_digits,
+        lambda value: f"{value} does not have exactly {num_digits} digits.",
+    )
 
-# custom validators
 
+# custom validators
 
+# 1973 01 13
 def dateMonthIsValid():
     """Validate that in a monthyear combination, the month is a valid month."""
     return make_validator(
         lambda value: int(str(value)[4:6]) in range(1, 13),
         lambda value: f"{str(value)[4:6]} is not a valid month.",
     )
 
+def dateDayIsValid():
+    """Validate that in a monthyearday combination, the day is a valid day."""
+    return make_validator(
+        lambda value: int(str(value)[6:]) in range(1, 31),
```

Comment:
```suggestion
        lambda value: int(str(value)[6:]) in range(1, 32),
```
python's `range` is end-exclusive. i only noticed this because the function above uses `range(1, 13)` for the month validation
![image](https://github.com/raft-tech/TANF-app/assets/111305129/b3339aef-33aa-47ee-875a-99e826a178cd)